coin_chk_lapack.m4

```diff
@@ -0,0 +1,195 @@
+
+###########################################################################
+#                       COIN_CHK_LAPACK                                   #
+###########################################################################
+
+# COIN_CHK_LAPACK([client packages])
+
+# This macro checks for a LAPACK library and adds the information necessary to
+# use it to the _LFLAGS, _CFLAGS, and _PCFILES variables of the client packages
+# passed as a space-separated list in parameter $1. These correspond to
+# Libs.private, Cflags.private, and Requires.private, respectively, in a .pc
+# file.
+
+# The algorithm first looks for command line parameters --with-lapack,
+# --with-lapack-lflags, and --with-lapack-cflags. If nothing is found,
+# system-specific default locations are checked, followed by some generic
+# defaults.  A link check is used to determine whether default locations
+# work and to determine the name mangling scheme of the Lapack library.
+
+AC_DEFUN([AC_COIN_CHK_LAPACK],
+[
+  AC_MSG_CHECKING(for LAPACK)
+
+dnl Make sure the necessary variables exist for each client package.
+  m4_foreach_w([myvar],[$1],
+    [AC_SUBST(m4_toupper(myvar)_CFLAGS)
+     AC_SUBST(m4_toupper(myvar)_LFLAGS)
+     AC_SUBST(m4_toupper(myvar)_PCFILES)
+    ])
+
+dnl Set up command line arguments with DEF_PRIM_ARGS.
+  AC_COIN_DEF_PRIM_ARGS([lapack],yes,yes,yes,no)
+
+dnl Assume the worst and persist in the face of failure.
+  coin_has_lapack=no
+  lapack_keep_looking=yes
+
+dnl Look for user-specified lapack flags and try them if present. Pay
+dnl attention if the user says 'no' on the command lin.
+  if test "$with_lapack" != "no" ; then
+    if test -n "$with_lapack_lflags" ; then
+      AC_COIN_TRY_LINK([dorhr_col],[$with_lapack_lflags],[],
+        [AC_MSG_RESULT([yes (user-specified)])
+	 lapack_keep_looking=no
+	 lapack_lflags=$with_lapack_lflags
+	 lapack_cflags=$with_lapack_cflags],
+	[AC_MSG_ERROR([Cannot link to user-specified Lapack.])])
+    fi
+  else
+    lapack_keep_looking=no
+    AC_MSG_RESULT([no (user-specified)])
+  fi
```

Comment:
The idea here is to pay attention to command line flags, if present. In particular, `--with-lapack=no` will be honored. If the user specifies `--with-lapack-lflags` or `--with-lapack-cflags`, that will be picked up and we'll try the link. `lapack_cflags` isn't actually used in the link test, but it's captured. I added `lapack_keep_looking` just to keep the code structure easy to see (didn't want to introduce a massive arm on an if-then-else statement).

Reply:
The TRY_LINK message was saying what it looks for using which linker flags, right? Don't you think that this was useful?

Reply:
It was definitely useful, just not very readable. I'll grab some comparative output and we can decide. Give me a bit of time for that.

Reply:
Here's the stdout output on my linux box configuring CoinUtils. With `TRY_LINK` messages turned off.
```
checking for LAPACK... yes (lapack.pc and blas.pc)
```
With `TRY_LINK` output turned on,
```
checking for LAPACK... checking for function dorhr_col in -lmkl_core -lmkl_intel_lp64 -lmkl_sequential -lm ... no
checking for function dorhr_col_ in -lmkl_core -lmkl_intel_lp64 -lmkl_sequential -lm ... no
checking for function DORHR_COL in -lmkl_core -lmkl_intel_lp64 -lmkl_sequential -lm ... no
checking for function DORHR_COL_ in -lmkl_core -lmkl_intel_lp64 -lmkl_sequential -lm ... no
checking for function dorhr_col_ in -lmkl_core -lmkl_intel_lp64 -lmkl_sequential -lm ... no
checking for function dorhr_col__ in -lmkl_core -lmkl_intel_lp64 -lmkl_sequential -lm ... no
checking for function DORHR_COL_ in -lmkl_core -lmkl_intel_lp64 -lmkl_sequential -lm ... no
checking for function DORHR_COL__ in -lmkl_core -lmkl_intel_lp64 -lmkl_sequential -lm ... no
yes (lapack.pc and blas.pc)
checking for function dorhr_col in -llapack  ... no  
checking for function dorhr_col_ in -llapack  ... yes
```
It's informative if you know what you're looking at (the attempt to figure out name mangling, which fails the first 8 times because ld can't find the MKL libraries) but not so much if you're unaware of name mangling and just want to know that some LAPACK was found.

The attempts to figure out name mangling do show in config.log, but (annoyingly) the final successful attempt is not shown, only the failures.

I turned `TRY_LINK` output off by default because it didn't seem to be used anywhere except `CHK_LAPACK`, so default off was minimal change. I can flip that, and explicitly set it to off in `CHK_LAPACK`. Then output will show if TRY_LINK is used elsewhere.

If what you're after is a bit more information, we could print the final results (link flags and name mangling) at the end.

Reply:
Na fine, maybe it was just interesting for me to see.

Just changing `AC_MSG_RESULT([yes (system special case)])` to print `$lapack_lflags`, too, would be nice.

Reply:
I've tweaked `CHK_LAPACK` to produce a message at the end showing what was found. If `coin_has_lapack = yes` the values of `lapack_lflags` and `lapack_pcfiles` are printed if they're not empty.